domain/state.go

```diff
@@ -218,12 +219,26 @@ type AtomicContext interface {
 // transaction. The mutex prevents data-races when the transaction is removed
 // from the context.
 type atomicContext struct {
-	context.Context
+	ctx context.Context
 
 	mu sync.Mutex
 	tx *sqlair.TX
 }
 
+// Context returns the context that the transaction was created with.
+func (c *atomicContext) Context() context.Context {
+	return c.ctx
+}
+
+// txn returns the transaction stored on the atomicContext.
+// This is used to run transactions within the context of the atomicContext.
+func (c *atomicContext) txn() *sqlair.TX {
```

Comment:
We could make this part of the AtomicContext interface. This would then allow us the opportunity to drop the if check in the Run function, as we would know that it could only be satisfied by something internal to this package.

Reply:
I would much rather take atomic context as a struct with a private field rather than this.